### PR TITLE
net-analyzer/linkchecker: add python3 support

### PR DESCRIPTION
**Pull request by hydrapolic**
https://bugs.gentoo.org/712958

Thanks @cjmayo

### CODE REVIEW

**Comment by ajakk** [comment]
Please rebase and force push to get more accurate CI results.

**Comment by a17r** [comment]
Who's going to maintain the live ebuild?

**Comment by hydrapolic** [comment]
> Who's going to maintain the live ebuild?

@cjmayo is doing upstream development, he stated in the bug he's using it, so I hope we can keep it up to date in portage too. If not, we'll drop it and just keep a released version or snapshot.

**Comment by a17r** [comment]
But `metadata.xml` is not listing anyone. Do you want to do it?

**Comment by hydrapolic** [comment]
@cjmayo is welcome to join too :)

File: net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+SLOT="0"
+IUSE="sqlite"
+
+RDEPEND="
+	dev-python/beautifulsoup[${PYTHON_USEDEP}]
+	dev-python/dnspython[${PYTHON_USEDEP}]
+	dev-python/pyxdg[${PYTHON_USEDEP}]
+	dev-python/requests[${PYTHON_USEDEP}]
+"
+DEPEND=""
```
Comment (mgorny) [inline net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild:33]:
```suggestion
```

File: net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6..8} )
```
Comment (mgorny) [inline net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild:6]:
```suggestion
PYTHON_COMPAT=( python3_{6..9} )
```

File: net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+RDEPEND="
+	dev-python/beautifulsoup[${PYTHON_USEDEP}]
+	dev-python/dnspython[${PYTHON_USEDEP}]
+	dev-python/pyxdg[${PYTHON_USEDEP}]
+	dev-python/requests[${PYTHON_USEDEP}]
+"
+DEPEND=""
+
+RESTRICT="test"
```
Comment (mgorny) [inline net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild:35]: RESTRICT goes below IUSE, also please add a comment why you're restricting them.

File: net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+S="${WORKDIR}/${PN}-${GIT_COMMIT}"
```
Comment (mgorny) [inline net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild:37]: S goes below SRC_URI. Also you probably don't want this in 9999 block.

File: net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+python_prepare_all() {
+	local PATCHES=(
+		"${FILESDIR}/${PN}-9.3-bash-completion.patch"
+	)
+
+	distutils-r1_python_prepare_all
+}
+
+python_install_all() {
+	DOCS=(
```
Comment (mgorny) [inline net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild:48]:
```suggestion
	local DOCS=(
```

File: net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild (hunk 2 above)
Comment (hydrapolic) [reply net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild:6]: Missing support in pyxdg: https://bugs.gentoo.org/735746

**Comment by hydrapolic** [comment]
> Thanks for willing to tackle this. A few comments nevertheless.

Thanks for the review @mgorny

**Comment by hydrapolic** [comment]
Ping @mgorny

File: net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild (hunk 2 above)
Comment (mgorny) [reply net-analyzer/linkchecker/linkchecker-10.0.0_pre20200729.ebuild:6]: Isn't this resolved?

**Review by mgorny — APPROVED** [review]
Sorry for the delay. I'll merge now.

**Comment by hydrapolic** [comment]
> Sorry for the delay. I'll merge now.

Thanks @mgorny. Hopefully that CI breakage related to pyxdg won't cause troubles (it appeared after adding py3.9).

**Pull request closed without merging**